### PR TITLE
fix(anta.tests): Cleaning up Interfaces tests module (VerifyLACPInterfacesStatus)

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactoring Interfaces tests module to address the following issues:

- https://github.com/aristanetworks/anta/issues/809
- https://github.com/aristanetworks/anta/issues/587
- https://github.com/aristanetworks/anta/issues/388

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/input_models/interfaces.py [hunk 1]
```diff
@@ -7,17 +7,33 @@
...
 
+    model_config = ConfigDict(extra="forbid")
     name: Interface
     """Interface to validate."""
-    status: Literal["up", "down", "adminDown"]
-    """Expected status of the interface."""
+    status: Literal["up", "down", "adminDown"] | None = None
+    """Expected status of the interface. Required field in the `VerifyInterfacesStatus` test."""
     line_protocol_status: Literal["up", "down", "testing", "unknown", "dormant", "notPresent", "lowerLayerDown"] | None = None
     """Expected line protocol status of the interface."""
```
Comment (carl-baillargeon) [inline anta/input_models/interfaces.py:24]: Should we update this and say this is an optional field in the `VerifyInterfacesStatus` test?

File: anta/input_models/interfaces.py [hunk 2]
```diff
@@ -7,17 +7,33 @@
...
+    portchannel: PortChannelInterface | None = None
+    """Port Channel in which the interface is bundled. Required field in the `VerifyLACPInterfacesStatus` test."""
```
Comment (carl-baillargeon) [inline anta/input_models/interfaces.py:26]: In EOS they put `Port-Channel` with an hyphen:
```
interface Port-Channel11
   description MLAG_DC1-LEAF1B_Port-Channel11
   switchport mode trunk
   switchport trunk group MLAG
```
```suggestion
    """Port-Channel in which the interface is bundled. Required field in the `VerifyLACPInterfacesStatus` test."""
```

File: anta/input_models/interfaces.py [hunk 3]
```diff
@@ -7,17 +7,33 @@
...
+        """Return a string representation of the InterfaceState model. Used in failure messages.
+
+        Examples
+        --------
+        - Interface: Ethernet1 Port Channel: Port-Channel100
+        - Interface: Ethernet1
+        """
+        base_string = f"Interface: {self.name}"
+        if self.portchannel is not None:
+            base_string += f" Port Channel: {self.portchannel}"
```
Comment (carl-baillargeon) [inline anta/input_models/interfaces.py:38]:
```suggestion
            base_string += f" Port-Channel: {self.portchannel}"
```

File: anta/tests/interfaces.py [hunk 4]
```diff
@@ -848,17 +848,22 @@ def test(self) -> None:
...
-    - Verifies that the interface is a member of the LACP port channel.
-    - Ensures that the synchronization is established.
-    - Ensures the interfaces are in the correct state for collecting and distributing traffic.
-    - Validates that LACP settings, such as timeouts, are correctly configured. (i.e The long timeout mode, also known as "slow" mode, is the default setting.)
+    This test performs the following checks for each specified interface:
+
+      1. Verifies that the interface is a member of the LACP port channel.
+      2. Ensures that the synchronization is established.
+      3. Ensures the interfaces are in the correct state for collecting and distributing traffic.
+      4. Validates that LACP settings, such as timeouts, are correctly configured. (i.e The long timeout mode, also known as "slow" mode, is the default setting.)
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:858]: Let's update this and specify exactly what we are testing per the code: `member_port_details = ["activity", "aggregation", "synchronization", "collecting", "distributing", "timeout"]`

Comment (carl-baillargeon) [reply anta/tests/interfaces.py:858]:
```
- Activity: Active LACP mode (initiates)
- Timeout: Long timeout (default)
- Aggregation: Port aggregable
- Synchronization: Port in sync with partner
- Collecting: Incoming frames aggregating
- Distributing: Outgoing frames aggregating
```

Comment (carl-baillargeon) [reply anta/tests/interfaces.py:858]: As discussed, let's also add `lacp_rate_fast: bool = False` field in the `InterfaceState` model to accomodate fast timeouts. Let's also put a note here in the docstring that we can provide the expected LACP timeout.

File: anta/input_models/interfaces.py (hunk 1 above)
Comment (vitthalmagadum) [reply anta/input_models/interfaces.py]: Updated.

File: anta/input_models/interfaces.py [hunk 5]
```diff
@@ -7,17 +7,33 @@
...
-    """Expected status of the interface."""
+    status: Literal["up", "down", "adminDown"] | None = None
+    """Expected status of the interface. Required field in the `VerifyInterfacesStatus` test."""
     line_protocol_status: Literal["up", "down", "testing", "unknown", "dormant", "notPresent", "lowerLayerDown"] | None = None
     """Expected line protocol status of the interface."""
+    portchannel: PortChannelInterface | None = None
+    """Port Channel in which the interface is bundled. Required field in the `VerifyLACPInterfacesStatus` test."""
+
+    def __str__(self) -> str:
+        """Return a string representation of the InterfaceState model. Used in failure messages.
```
[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/input_models/interfaces.py]: Updated.

File: anta/input_models/interfaces.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/input_models/interfaces.py]: Updated.

File: anta/input_models/interfaces.py (hunk 3 above)
Comment (vitthalmagadum) [reply anta/input_models/interfaces.py]: Updated.

File: anta/tests/interfaces.py (hunk 4 above)
Comment (vitthalmagadum) [reply anta/tests/interfaces.py]: Updated. Thanks for catching this!!

File: anta/tests/interfaces.py [hunk 6]
```diff
@@ -848,17 +848,26 @@ def test(self) -> None:
...
+    """Verifies the Link Aggregation Control Protocol (LACP) status of the interface.
 
-    - Verifies that the interface is a member of the LACP port channel.
-    - Ensures that the synchronization is established.
-    - Ensures the interfaces are in the correct state for collecting and distributing traffic.
-    - Validates that LACP settings, such as timeouts, are correctly configured. (i.e The long timeout mode, also known as "slow" mode, is the default setting.)
+    This test performs the following checks for each specified interface:
+
+      1. Verifies that the interface is a member of the LACP port channel.
+      4. Validates that LACP settings are correctly configured to ensure reliable link aggregation. This includes:
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:856]:
```suggestion
      2. Verifies LACP port states and operational status:
```

File: anta/tests/interfaces.py [hunk 7]
```diff
@@ -848,17 +848,26 @@ def test(self) -> None:
...
+        - Activity: Active LACP mode (initiates)
+        - Timeout: Short (Fast Mode), Long (Slow Mode) as specified.
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:858]:
```suggestion
        - Timeout: Short (Fast Mode), Long (Slow Mode - default)
```

File: anta/tests/interfaces.py [hunk 8]
```diff
@@ -848,17 +848,26 @@ def test(self) -> None:
...
+        - Aggregation: Port aggregable
+        - Synchronization: Port in sync with partner
+        - Collecting: Incoming frames aggregating
+        - Distributing: Outgoing frames aggregating
 
     Expected Results
     ----------------
-    * Success: The test will pass if the provided interfaces are bundled in port channel and all specified parameters are correct.
-    * Failure: The test will fail if any interface is not bundled in port channel or any of specified parameter is not correct.
+    * Success: If the provided interfaces are bundled in port channel and all specified parameters are correct.
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:866]:
```suggestion
    * Success: Interface is bundled and all LACP states match expected values for both actor and partner
```

File: anta/tests/interfaces.py [hunk 9]
```diff
@@ -848,17 +848,26 @@ def test(self) -> None:
...
+    * Failure: If any of the following occur:
+        - The specified interface and port channel is not configured.
+        - The specified interface is not bundled in given port channel.
+        - The specified parameters does not match the expected operational state for any interface.
```
Comment (carl-baillargeon) [inline anta/tests/interfaces.py:870]:
```suggestion
        - Interface or port channel is not configued.
        - Interface is not bundled in port channel.
        - Actor or partner port LACP states don't match expected configuration.
        - LACP rate (timeout) mismatch when fast mode is configued.
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**